### DISCUSSION TITLE
Global Pooling

### DISCUSSION THREAD

**Discussion by hadarshavit** [pyg-team/pytorch_geometric#5380] [Q&A] [answered]
Hi, I'm using GNNs for graph classification. Therefore, I need to aggregate the node embeddings. In PyG there are the "basic" pooling layers: max, mean, add.
I'm looking for more advanced pooling layers such as global attention pooling (like in [https://arxiv.org/abs/1511.05493](https://arxiv.org/abs/1511.05493)), but could not find an implementation of it.
Is there an implementation of more advanced pooling layers in PyG?
Thanks!

**Comment by EdisonLeeeee** [comment, marked as answer]
You can take a look at [here](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#pooling-layers), where some advanced pooling layers were introduced.

**Reply by rusty1s** [reply]
Please also take a look at [this](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#aggregation-operators). All aggregation modules can also be used for global pooling.

**Reply by hadarshavit** [reply]
@rusty1s Thanks, is there a tutorial on how to use the aggregations as global pooling?

**Reply by rusty1s** [reply]
I will add this piece of information to the documentation. In general, it is as easy as doing:
```python
aggr = MeanAggregation(...)

aggr(x, batch)
```
